internal/auth/docker.go

```diff
@@ -0,0 +1,70 @@
+package auth
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"path/filepath"
+)
+
+// DockerConfig represents the Docker config.json structure
+type DockerConfig struct {
+	Auths map[string]DockerAuth `json:"auths"`
+}
```

Comment:
Please use `viper` for the config - may as well since it's in the repo, there is an example in `pkg/config/config.go`